packages/devtools_app/lib/src/flutter/extent_delegate_list.dart

```diff
@@ -0,0 +1,464 @@
+        constraints.scrollOffset + constraints.cacheOrigin;
+    assert(scrollOffset >= 0.0);
+    final double remainingExtent = constraints.remainingCacheExtent;
+    assert(remainingExtent >= 0.0);
+    final double targetEndScrollOffset = scrollOffset + remainingExtent;
+
+    final int firstIndex =
+        _extentDelegate.minChildIndexForScrollOffset(scrollOffset);
+    final int targetLastIndex = targetEndScrollOffset.isFinite
+        ? _extentDelegate.maxChildIndexForScrollOffset(targetEndScrollOffset)
+        : null;
+
+    if (firstChild != null) {
+      final int leadingGarbage = _calculateLeadingGarbage(firstIndex);
+      final int trailingGarbage = _calculateTrailingGarbage(targetLastIndex);
+      collectGarbage(leadingGarbage, trailingGarbage);
+    } else {
+      collectGarbage(0, 0);
+    }
+
+    if (firstChild == null) {
+      if (!addInitialChild(
+          index: firstIndex,
+          layoutOffset: _extentDelegate.layoutOffset(firstIndex))) {
+        // There are either no children, or we are past the end of all our children.
+        // If it is the latter, we will need to find the first available child.
+        double max;
+        if (childManager.childCount != null) {
+          // TODO(jacobr): is this correct?
+          // This matches the logic from sliver_fixed_extent_list but it seems
+          // a little odd. This is only right if the childManager contains all
+          // children added to the list view.
+          max = _extentDelegate.layoutOffset(childManager.childCount);
+        } else if (firstIndex <= 0) {
+          max = 0.0;
+        } else {
+          // We will have to find it manually.
+          int possibleFirstIndex = firstIndex - 1;
+          while (possibleFirstIndex > 0 &&
+              !addInitialChild(
+                index: possibleFirstIndex,
+                layoutOffset: _extentDelegate.layoutOffset(possibleFirstIndex),
+              )) {
+            possibleFirstIndex -= 1;
+          }
+          max = _extentDelegate.layoutOffset(possibleFirstIndex);
+        }
+        geometry = SliverGeometry(
+          scrollExtent: max,
+          maxPaintExtent: max,
+        );
+        childManager.didFinishLayout();
+        return;
+      }
+    }
+
+    RenderBox trailingChildWithLayout;
+
+    for (int index = indexOf(firstChild) - 1; index >= firstIndex; --index) {
+      final RenderBox child =
+          insertAndLayoutLeadingChild(buildChildConstraints(index));
+      if (child == null) {
+        // Items before the previously first child are no longer present.
+        // Reset the scroll offset to offset all items prior and up to the
+        // missing item. Let parent re-layout everything.
+        geometry = SliverGeometry(
+            scrollOffsetCorrection: _extentDelegate.layoutOffset(index));
+        return;
+      }
+      final SliverMultiBoxAdaptorParentData childParentData =
+          child.parentData as SliverMultiBoxAdaptorParentData;
```

Comment:
same question here and elsewhere about whether both the type and the cast are necessary

Reply:
not needed. Removed. this code is from Flutter so it is  has a few too many type checks.